components/supervisor/pkg/supervisor/services.go

```diff
@@ -777,6 +779,44 @@ func (ss *ControlService) CreateSSHKeyPair(context.Context, *api.CreateSSHKeyPai
 	}, err
 }
 
+// CreateDebugEnv creates a debug workspace envs
+func (c *ControlService) CreateDebugEnv(ctx context.Context, req *api.CreateDebugEnvRequest) (*api.CreateDebugEnvResponse, error) {
+	var envs []string
+	for _, env := range os.Environ() {
+		if env == "" {
+			continue
+		}
+		parts := strings.SplitN(env, "=", 2)
+		key := parts[0]
+		if strings.HasPrefix(key, "THEIA_") ||
+			strings.HasPrefix(key, "GITPOD_") ||
+			// TODO IDE - get rid of env vars in images, use supervisor api as a mediator to support many IDEs running in the same worksapce?
+			// TODO PATH - use well defined locations to pick up binaries, i.e. /ide/bin or /ide-desktop/bin in supervisor?
```

Comment:
Pudong suggested to put it under /usr/bin instead

Reply:
@iQQBot I realized `/usr/bin` is not going to work, we are not prepending, but appending. Right now a user can bring own version of `code` cli and so on. We had problems with that in the past when users are running another version of VS Code in parallel. I think to avoid breaking we will need to append our cli from another location still.